src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -380,7 +380,11 @@ namespace winrt::TerminalApp::implementation
         auto loadedRevoker{ dialog.Loaded(winrt::auto_revoke, themingLambda) }; // if it's not yet in the tree
 
         // Display the dialog.
-        co_return co_await dialog.ShowAsync(Controls::ContentDialogPlacement::Popup);
+        const auto result = co_await dialog.ShowAsync(Controls::ContentDialogPlacement::Popup);
+        // GH#12622: After the dialog is displayed, always clear it out. If we
+        // don't, we won't be able to display another!
+        _dialog = nullptr;

```

Comment:
Above it says
```cpp
_dialog = dialog;
```

To make this code robust against exceptions and future early returns I'd suggest changing it to:
```cpp
_dialog = dialog;
const auto cleanup = wil::scope_exit([this]() {
    _dialog = nullptr;
});
```